src/plugins/intel_gpu/src/runtime/layout.cpp

```diff
@@ -0,0 +1,172 @@
+
+    if (dims[0] != size.group[0]) {
+        throw std::runtime_error("groups mismatch: " + std::to_string(dims[0]) + " vs " + std::to_string(size.group[0]));
+    }
+    return dims[0];
+}
+
+tensor::value_type layout::ofm() const {
+    if (!format::is_weights_format(format)) {
+        throw std::logic_error("[GPU] can't get OFM dimension for data layout");
+    }
+    auto dims = get_dims();
+    size_t dim_idx = format::is_grouped(format) ? 1 : 0;
+
+    return dims[dim_idx];
+}
+
+tensor::value_type layout::ifm() const {
+    if (!format::is_weights_format(format)) {
+        throw std::logic_error("[GPU] can't get IFM dimension for data layout");
+    }
+    auto dims = get_dims();
+    size_t dim_idx = format::is_grouped(format) ? 2 : 1;
+    return dims[dim_idx];
+}
+
+static format get_default_format(size_t rank, bool is_weights, bool is_grouped) {
+    auto default_fmt = cldnn::format::bfyx;
+    if (is_weights) {
+        if (is_grouped) {
+            if (rank == 5) {
+                default_fmt = cldnn::format::goiyx;
+            } else if (rank == 6) {
+                default_fmt = cldnn::format::goizyx;
+            }
+        } else {
+            if (rank == 4) {
+                default_fmt = cldnn::format::oiyx;
+            } else if (rank == 5) {
+                default_fmt = cldnn::format::oizyx;
+            }
+        }
+    } else {
+        if (rank == 5) {
+            default_fmt = cldnn::format::bfzyx;
+        } else if (rank == 6) {
+            default_fmt = cldnn::format::bfwzyx;
+        }
+    }
+
+    return default_fmt;
+}
+std::vector<tensor::value_type> layout::get_dims() const {
+    auto default_fmt = get_default_format(format.dimension(), format::is_weights_format(format), format::is_grouped(format));
+    return size.sizes(default_fmt);
+}
+
+std::vector<tensor::value_type> layout::get_padded_dims() const {
+    auto default_fmt = get_default_format(format.dimension(), format::is_weights_format(format), format::is_grouped(format));
+    auto padded_size = size.add(data_padding.lower_size()).add(data_padding.upper_size());
+    return padded_size.sizes(default_fmt);
+}
+
+static format to_weights_format(format f, bool is_grouped) {
+    if (format::is_weights_format(f))
+        return f;
+
+    switch (f) {
+        case format::bfyx:
+            return format::oiyx;
+        case format::fyxb:
+            return format::iyxo;
+        case format::byxf:
+            return format::oyxi;
+        case format::yxfb:
+            return format::yxio;
+        case format::bfzyx:
+            return is_grouped ? format::goiyx : format::oizyx;
+        case format::bfwzyx: {
+            if (!is_grouped)
+                throw std::runtime_error("Invalid conversion of data format to weights format. bfwzyx can't be non-grouped as 4D spatials are not supported");
+            return format::goizyx;
+        }
+        case format::bs_xs_xsv8_bsv8:
+            return format::os_i_osv8__ai8;
```

Comment:
Is this mapping or naming of format `os_i_osv8__ai8` correct? I have a feeling that `bs_xs_xsv8_bsv8` and `os_i_osv8__ai8` are not exactly the same, since `os_i_osv8__ai8` means blocking by `o` dimension and aligning `i` dimension by 8, but `bs_xs_xsv8_bsv8` means blocking by `o` and `i` dimensions

Reply:
That's a good point. For now I've aligned cldnn::format names with corresponding kernel_selector format name. FC kernel request WeightsLayout::os_i_osv8__ai8 weights format which is then converted to bs_xs_xsv8_bsv8. So this part just adds backward conversion between these formats. But I think we need to revise format names to ensure they are valid.